rust/src/tests/truncate.rs

```diff
@@ -0,0 +1,80 @@
+use std::{fs::File, io::Write};
+
+use nix::{errno::Errno, sys::stat::lstat, unistd::truncate};
+use rand::random;
+
+use crate::{
+    runner::context::{FileType, SerializedTestContext},
+    test::TestContext,
+    tests::{assert_ctime_changed, assert_ctime_unchanged},
+};
+
+// tests/truncate/00.t
+
+crate::test_case! {
+    /// truncate extends if the file is empty
```

Comment:
The description is inaccurate, because this test covers both extending and shrinking.  The next test case covers shrinking a dense file, whereas this test case covers shrinking a sparse file.